pytorch_lightning/callbacks/pruning.py

```diff
@@ -196,17 +199,18 @@ def __init__(
                 " HINT: if passing a `BasePruningMethod`, pass the the class, not an instance"
             )
 
-        if use_global_unstructured and pruning_fn.PRUNING_TYPE != "unstructured":
+        # need to ignore typing here since pytorch base class does not define the PRUNING_TYPE attribute
+        if use_global_unstructured and pruning_fn.PRUNING_TYPE != "unstructured":  # type: ignore
             raise MisconfigurationException(
-                'Only the "unstructured" PRUNING_TYPE is supported with `use_global_unstructured=True`.'
+                'Only the "unstructured" PRUNING_TYPE is supported with `use_global_unstructured=True`.'  # type: ignore
```

Comment:
why is this type ignore in the middle of this string here?

Reply:
See the comment right above: "need to ignore typing here since pytorch base class does not define the PRUNING_TYPE attribute"